File: Lists/ProbList101/P4414/src/Main.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] arr = new int[]{sc.nextInt(), sc.nextInt(), sc.nextInt()};
        Arrays.sort(arr);
        String s = sc.next();
        if (s.equals("ABC")) System.out.println(arr[0] + " " + arr[1] + " " + arr[2]);
        else if (s.equals("ACB")) System.out.println(arr[0] + " " + arr[2] + " " + arr[1]);
        else if (s.equals("BAC")) System.out.println(arr[1] + " " + arr[0] + " " + arr[2]);
        else if (s.equals("BCA")) System.out.println(arr[1] + " " + arr[2] + " " + arr[0]);
        else if (s.equals("CAB")) System.out.println(arr[2] + " " + arr[0] + " " + arr[1]);
        else if (s.equals("CBA")) System.out.println(arr[2] + " " + arr[1] + " " + arr[0]);
    }
}

File: Lists/ProbList101/P4414/src/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
    int arr[4];
    cin >> arr[0] >> arr[1] >> arr[2];
    sort(arr, arr+3);
    char ch;
    for(int i = 0; i < 3; i++)
	{
		cin >> ch;
		if(ch == 'A') cout << arr[0] << ' ';
		if(ch == 'B') cout << arr[1] << ' ';
		if(ch == 'C') cout << arr[2] << ' ';
	}

    return 0;
}